### ISSUE TITLE
would you consider making ComparisonConfig immutable?

### ISSUE THREAD

**Issue by drdamour** [GregFinzer/Compare-Net-Objects#305]
totally understand this would be a backwards breaking change. 

We have quite a few ComparisonConfig constants for various common cases, but it's always a bit of a worry passing these around or referencing them that they will be mutated.  To work around this we regenerate them on every get, but it'd be slick if the lib itself handled this.  Method like CustomPropertyConverter<> would switch to return a copy instead of void.  that would be quite a backwards breaking change.

if you were conformable going that far, an even further step would be to make it a record type, which would bring with it the with {} syntax.  Doing this doesn't force you into any modern .net as the IL ends up being class still (with some annotations) AFAIK. but it isn't totally free of headache as you do have to have your lib be lang ver 9.

if u were interested, we'd submit a PR to this effect, but not worth the effort only for it to be rejected

**Comment by GregFinzer** [comment]
Probably tour best option is to create a fork rather than to create a breaking change.  There are over 24 Million downloads now.

**Comment by drdamour** [comment]
yeah it would certainly require a v5 (+1 major ver) at a minimum, dunno if you had a bucket of breaking changes stuff you're collecting.

**Comment by GregFinzer** [comment]
No, there currently are no open issues.
